src/forms/ToggleSwitch.jsx

```diff
@@ -0,0 +1,123 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import cx from 'classnames';
+
+import { C_TEXT_PRIMARY, C_TEXT_SECONDARY, C_TEXT_HINT } from 'swarm-constants/dist/js/colorConstants.js';
+
+import Button from './Button';
+import Icon from '../media/Icon';
+import withToggleControl from '../utils/WithToggleControl';
+
+export const TOGGLE_SWITCH_CLASS = 'toggleSwitch';
+export const TOGGLE_SWITCH_ACTIVE_CLASS = `${TOGGLE_SWITCH_CLASS}--active`;
+export const TOGGLE_SWITCH_DISABLED_CLASS = `${TOGGLE_SWITCH_CLASS}--disabled`;
+
+export const KNOB_CLASS = `${TOGGLE_SWITCH_CLASS}-knob`;
+export const KNOB_ACTIVE_CLASS = `${KNOB_CLASS}--active`;
+export const KNOB_DISABLED_CLASS = `${KNOB_CLASS}--disabled`;
+
+export const LABEL_CLASS = `${TOGGLE_SWITCH_CLASS}-label`;
+
+/**
+ * @module ToggleSwitch
+ */
+class ToggleSwitch extends React.Component {
+	constructor (props) {
+		super(props);
+
+		this.toggle = this.toggle.bind(this);
+	}
+
+	toggle() {
+		!this.props.disabled && this.props.toggleActive();
+	}
+
+	componentDidUpdate(prevProps) {
+		if (prevProps !== this.props) {
```

Comment:
This is getting into the gray area of my React knowledge... Couldn't multiple props have changed? I guess in this component we'd only expect `isActive` to be changing, but I still wonder if `prevProps.isActive !== props.isActive` would be mo' better on any level?

Reply:
We don't _currently_ have any other props that might change, but I'll make it more specific just incase we ever do :)